lib/Backend/Vault/Kv.hs

```diff
@@ -54,6 +55,66 @@ data VaultKvBackend =
   }
   deriving stock (Show)
 
+-- | Errors that can be thrown in Vault backend.
+data VaultError
+  = ServantError ClientError
+  | FieldMetadataNotFound EntryPath FieldKey
+  | BadFieldNameError BadFieldName
+  | BadMasterFieldName Text BadFieldName
+  | BadEntryTagError BadEntryTag
+  | CofferSpecialsNotFound EntryPath
+  | BadCofferSpecialsError Text
+
+instance Buildable VaultError where
+  build = \case
+    ServantError (FailureResponse request response) ->
+      unlinesF @_ @Builder
+        [ "Request:"
+        , indentF 2 ((build . show) request)
+        , "failed with response:"
+        , indentF 2 ((build . show) response)
+        ]
+    ServantError (DecodeFailure body response) ->
+      unlinesF @_ @Builder
+        [ "The body could not be decoded at the expected type."
+        , "Body: " <> build body
+        , "Response:"
+        , indentF 2 ((build . show) response)
+        ]
+    ServantError (UnsupportedContentType mediatype response) ->
+      unlinesF @_ @Builder
+        [ "The content-type '" <> (build . show) mediatype <> "' of the response is not supported."
+        , "Response:"
+        , indentF 2 ((build . show) response)
+        ]
+    ServantError (InvalidContentTypeHeader response) ->
+      unlinesF @_ @Builder
+        [ "The content-type header is invalid."
+        , "Response:"
+        , indentF 2 ((build . show) response)
+        ]
+    ServantError (ConnectionError exception) ->
+      unlinesF @_ @Builder
+        [ "Connection error. No response was received."
+        , (build . show) exception
+        ]
+    FieldMetadataNotFound entryPath fieldName ->
+      "Could not find coffer metadata for field '" +| fieldName
+        |+ "' at '" +| entryPath |+ "'"
+    BadFieldNameError err -> build err
```

Comment:
This is not real an error specific to Vault, we'll end up with duplication in the Pass backend.